src/main/java/task/Duke.java

```diff
@@ -0,0 +1,143 @@
+            String command = TaskManager.getFirstWordFromCommand(input);
+
+            switch (command) {
+            case "bye":
+                Printer.printByeMessage();
+                isOver = true;
+                break;
+            case "list":
+                Printer.printLineSeparator();
+                System.out.println("Here are the tasks in your list:");
+
+                //Lists down all the tasks added along with its status
+                for (int i = 0; i < count; i++) {
+                    System.out.println((i + 1) + ". " + list.get(i));
+                }
+
+                Printer.printLineSeparator();
+                break;
+            case "delete":
+                try {
+                    int indexOfDelete = TaskManager.getIndex(input);
+
+                    Printer.printLineSeparator();
+                    System.out.println("Noted. I've deleted this task:\n" + list.get(indexOfDelete) + "\nNow you have " + (count - 1) + " tasks in the list.");
+                    Printer.printLineSeparator();
+
+                    list.remove(indexOfDelete);
+                    count--;
+                    storage.saveData(list);
+                } catch (IndexOutOfBoundsException | NumberFormatException e) {
+                    Error.showDeleteFormatError();
+                }
+                break;
+            case "done":
+                try {
+                    //Extracts the index number from the text and changes status of the task
+                    int indexOfDone = TaskManager.getIndex(input);
+                    list.get(indexOfDone).markAsDone();
+
+                    Printer.printLineSeparator();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(list.get(indexOfDone));
+                    Printer.printLineSeparator();
+                    storage.saveData(list);
+                } catch (IndexOutOfBoundsException | NumberFormatException e) {
+                    Error.showDoneFormatError();
+                }
+                break;
+            case "todo":
+                try {
+                    Task todo = TaskManager.getTodoDetails(input);
+
+                    Printer.printLineSeparator();
+                    System.out.println("Got it. I've added this task:\n" + todo + "\nNow you have " + count + " tasks in the list.");
+                    Printer.printLineSeparator();
+                    storage.saveData(list);
+                } catch (DukeEmptyTaskDescriptionException e) {
+                    Error.showTaskDescriptionError();
+                }
+                break;
+            case "deadline":
+                try {
+                    Task deadline = TaskManager.getDeadlineDetails(input);
+
+                    Printer.printLineSeparator();
+                    System.out.println("Got it. I've added this task:\n" + deadline + "\nNow you have " + count + " tasks in the list.");
```

Comment:
Awesome statement to indicate that the task has been added! Consider saving certain strings as constants to make your print statements more concise :)